contracts/DefaultStrategy.sol

```diff
@@ -422,83 +349,70 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
+     * @return The length of active groups.
      */
     function getGroupsLength() external view returns (uint256) {
-        return activeGroups.length();
-    }
-
-    /**
-     * @notice Returns the active group at index.
-     * @return The active group.
-     */
-    function getGroup(uint256 index) external view returns (address) {
-        return activeGroups.at(index);
-    }
-
-    /**
-     * @notice Returns whether active groups contain group.
-     * @return Whether or not is active group.
-     */
-    function groupsContain(address group) external view returns (bool) {
-        return activeGroups.contains(group);
+        return activeGroups.getNumElements();
     }
 
     /**
-     * @notice Returns the length of deprecated groups.
-     * @return The length of deprecated groups.
+     * @notice Returns previous and next address of key.
+     * @param key The key of searched group.
+     * @return previousAddress The previous address.
+     * @return nextAddress The next address.
      */
-    function getDeprecatedGroupsLength() external view returns (uint256) {
-        return deprecatedGroups.length();
+    function getGroupPreviousAndNext(address key)
+        external
+        view
+        returns (address previousAddress, address nextAddress)
+    {
+        (, previousAddress, nextAddress) = activeGroups.get(key);
     }
 
     /**
-     * @notice Returns the deprecated group at index.
-     * @return The deprecated group.
+     * @notice Returns head and previous address of head.
+     * @return head The head of groups.
+     * @return previousAddress The previous address.
      */
-    function getDeprecatedGroup(uint256 index) external view returns (address) {
-        return deprecatedGroups.at(index);
+    function getGroupsHead() external view returns (address head, address previousAddress) {
+        head = activeGroups.getHead();
+        (, previousAddress, ) = activeGroups.get(head);
     }
 
     /**
-     * @notice Returns whether deprecated groups contain group.
-     * @return The group.
+     * @notice Returns tail and next address of tail.
+     * @return head The tail of groups.
+     * @return nextAddress The previous address.
      */
-    function deprecatedGroupsContain(address group) external view returns (bool) {
-        return deprecatedGroups.contains(group);
+    function getGroupsTail() external view returns (address head, address nextAddress) {
+        head = activeGroups.getTail();
+        (, nextAddress, ) = activeGroups.get(head);
     }
 
     /**
-     * @notice Returns the list of deprecated groups.
-     * @return The list of deprecated groups.
+     * @notice Returns whether active groups contain group.
+     * @return Whether or not is active group.
      */
-    function getDeprecatedGroups() external view returns (address[] memory) {
-        return deprecatedGroups.values();
+    function groupsContain(address group) external view returns (bool) {
```

Comment:
Would be good to be more descriptive than just using underlying data structure terms, e.g. `isActive`.